features/update/basic_async/feature.ts

```diff
@@ -0,0 +1,48 @@
+import { WorkflowUpdateFailedError } from '@temporalio/client';
+import { Feature } from '@temporalio/harness';
+import * as wf from '@temporalio/workflow';
+import * as assert from 'assert';
+
+const myUpdate = wf.defineUpdate<string, [string]>('myUpdate');
+
+/**
+ * A workflow with an update and an update validator. If accepted, the update
+ * makes a change to workflow state. The workflow does not terminate until such
+ * a change occurs.
+ */
+export async function workflow(): Promise<string> {
+  let state = '';
+  const handler = (arg: string) => {
+    state = arg;
+    return 'update-result';
+  };
+  const validator = (arg: string) => {
+    if (arg == 'invalid-arg') {
+      throw new Error('Invalid Update argument');
+    }
+  };
+  wf.setHandler(myUpdate, handler, { validator });
+  await wf.condition(() => state != '');
+  return state;
+}
+
+export const feature = new Feature({
```

Comment:
I'm not sure what is the intended scope of this test. Seems like it is testing two distinct scenarios.

Please add a README.md file for this feature.

Reply:
`update/basic_async/README.md` is in `main` (added in a recently-merged PR). I've improved it in this branch.

We're deliberately testing validation in `update/basic` and `update/basic_async`. An alternative would be to have `update/basic`, `update/validation`, `update/basic_async`, and `update/validation_async`. We can move to that if we want. As things stand the non-TS `update/basic/*` are not implementing their spec since they need to test validation.

Reply:
Don't need to split. As long as the README file clearly define the scope of that test, I don't really mind that much the fact that a single feature test covers two tightly related scenarios.